workflow/torch/module_compose.py

```diff
@@ -20,6 +20,20 @@ def __init__(self, *modules_and_functions):
             )
         ])
 
+        self.parameter_list = nn.ParameterList([
+            parameter
+            for parameter in self.modules_and_functions
+            if isinstance(parameter, nn.Parameter)
+        ] + [
+            parameter_or_tuple[0]
+            for parameter_or_tuple in self.modules_and_functions
+            if (
+                type(parameter_or_tuple) is tuple and
+                isinstance(parameter_or_tuple[0], nn.Parameter)
```

Comment:
I'm not super well versed how it works in practice, does all elements of the tuple need to be a nn.Parameter?

Reply:
No only the first should be, the second should be a function

Reply:
aah, wait, forgot that you're pushing it into nn.ParameterList, you can divert the responsibility to not have a corrupt state to that